src/Field/Definition/ChoiceField.php

```diff
@@ -80,6 +104,23 @@ public function validateData (ComponentContext $context, array $contentPath, mix
 			$data = $context->normalizeOptionalString($data);
 		}
 
+		if (\is_array($data) && $this->allowMultiselect && (null !== $this->minimumNumberOfOptions || null !== $this->maximumNumberOfOptions))
```

Comment:
We never validate for an array.. I think we should separate the method into two main strains with an early `if ($this->multiselect)`

Reply:
That's true. We've also never validated empty single-selects correctly (missing `NotBlank`). I've added both and split up the validation.

It's much better that way. I've also checked whether it's possible to keep the call to `$this->choices->getValidationConstraints(…)` in the common code, but that's a bit harder as the single-select transforms `$data`, while the multi-select doesn't.